### ISSUE TITLE
DjangoCMS with Aldryn Newsblog urls clash - issue adding new non-blog pages

### ISSUE THREAD

**Issue by dmk12** [divio/aldryn-newsblog#372]
I am using Aldryn Newsblog and Aldryn FAQ in my DjangoCMS project. When trying to add a new page that is not a blog page, but just a normal CMS page, I get the following error: problem adding new page to Django CMS

![screen shot 2016-03-14 at 10 28 14](https://cloud.githubusercontent.com/assets/2589523/13874964/e51155ec-ecf2-11e5-8e02-978edd109d68.png)

Seems like Aldryn Newsblog rewrites the urls and sets itself as the default root.

Here's my urls.py file:

```
from __future__ import absolute_import, print_function, unicode_literals

from cms.sitemaps import CMSSitemap
from django.conf import settings
from django.conf.urls import *  # NOQA
from django.conf.urls.i18n import i18n_patterns
from django.contrib import admin
from django.contrib.staticfiles.urls import staticfiles_urlpatterns

admin.autodiscover()

urlpatterns = patterns('',
    url(r'^admin/', include(admin.site.urls)),  # NOQA
    url(r'^sitemap\.xml$', 'django.contrib.sitemaps.views.sitemap',
        {'sitemaps': {'cmspages': CMSSitemap}}),
    url(r'^select2/', include('django_select2.urls')),
    url(r'^', include('cms.urls')),
)

# This is only needed when using runserver.
if settings.DEBUG:
    urlpatterns = patterns('',
        url(r'^media/(?P<path>.*)$', 'django.views.static.serve',  # NOQA
            {'document_root': settings.MEDIA_ROOT, 'show_indexes': True}),
        ) + staticfiles_urlpatterns() + urlpatterns  # NOQA
```

How to I tweak it to account for Aldryn Newsblog? 

Here is the corresponding StackOverflow question, earn +50 rep by answering: 
http://stackoverflow.com/questions/35986368/djangocms-with-aldryn-newsblog-urls-clash-issue-adding-new-non-blog-pages

**Comment by victorevector** [comment]
Hi @dmk12 @mkoistinen ! 

I am also suffering from the same problem. I tried the solution in the stack overflow and it only partially worked. It allowed me to add new pages but did not fix the problem of accessing those pages. 

I found something interesting while trying to debug. In the photo below, url patterns 20-36 represent the Aldryn_NewsBlog urls (red). If you compare them to 7-17, url patterns for Aldryn_Events, you'll see that Aldryn_events makes sure to prefix it's urls with /events/ while Aldryn_NewsBlog does not. 

<img width="1229" alt="screen shot 2016-06-03 at 2 16 48 pm" src="https://cloud.githubusercontent.com/assets/10040882/15793215/69155f38-2997-11e6-96cc-a34ca0277acd.png">

Also of note is url pattern 37 (blue), which represents the CMS pages. It has the same regex pattern as Aldryn_Newsblog (20). I'm guessing that this relates to the conflict.

I'm guessing that a permanent fix would include making sure that Aldryn_NewsBlog add a prefix to its urls just like Aldryn_Events. I'm currenly stuck as to how one would do this.

Thoughts?

**Comment by mkoistinen** [comment]
Very curious. On which page (by slug) is your newsblog-instance attached? From the above, it looks like either you've attached the Aldryn News & Blog to your root CMS page, or perhaps to a page which has an empty slug.

The pattern on lines 20-36 are normal and will work fine as they should be made unique by the slug of the page you've attached the blog to.

Can you provide more information here?

I've never seen happen anywhere, and I wonder if there's some use-case that we didn't imagine.

**Comment by victorevector** [comment]
@mkoistinen My newblog-instance is saved to the root CMS page. The root CMS page has a slug, "home" that isn't acknowledged by the system. For example, going to "/home" will not direct you to the page. Only "/" will. 

I've attached it to another page that is not the root CMS page and you're right, the blog does adopt the page's slug.

**Comment by mkoistinen** [comment]
This is the correct behavior of both django CMS and the Aldryn News & Blog. I recommend that if you'd prefer to have the articles prefixed with a slug like "articles", then please create a page in the CMS with a slug "articles" and attach the News & Blog app to that page. This will allow you to also have other pages in the CMS as desired.

**Comment by sunmoonone** [comment]
I am also suffering from the same problem. I follows the tutorials on django-cms docs and everything works fun in the beginning: I can run server login and add pages. But when it comes to integrate with aldryn-newsblog, after I installed this addon and modified settings.py and restarted my local server. I got  `404 error` when try to access the root location, and  got `'WSGIRequest' object has no attribute 'user'` when try to access /en/ location.

I guess the cause is the different version of django in my virtul env. The initial version is 1.11 before aldryn-newsblog got installed, then it downgraded to version 1.9.13. But when I forcely reinstalled django to 1.10.8 another problem comes: `content field blablabla non-editable field`. 
So newsblog requires django<1.10 and the `404 error` is a bug in django-1.9.13 I think.
